### DISCUSSION TITLE
Feature Request: Integration with Glue Data Catalog

### DISCUSSION THREAD

**Discussion by nicor88** [duckdb/duckdb#11963] [Ideas]
DuckDb allows to read files from S3, that is pretty awesome.

In my setup the S3 table prefixes can change really often and variably. The single source of truth for where a specific table is located it's contained in Glue Data Catalog, that allow to make the table queryable via Athena/Trino/Spark/Redshift Spectrum.

It will be amazing to have duck db integration with a glue catalog (or catalog in general), that allow to pick automatically a specific s3 location for a specific table e.g.
```
select * from catalog('glue','my_db', 'my_table'); 
```
Under the hood I expect that based on what is contained in the catalog metadata information, the right under the hood functions are invoked in duckdb e.g. read_parquet/ iceberg_scan

**Comment by alberttwong** [comment]
I saw that it supports Unity Catalog (https://github.com/duckdb/uc_catalog?tab=readme-ov-file#unity-catalog-extension) and Delta Lake

**Reply by nicor88** [reply]
Interesting, indeed the catalog feature seems to be more an extension.

**Comment by alberttwong** [comment]
https://motherduck.canny.io/feature-requests/p/trying-to-query-iceberg-delta-and-or-hudi-data-that-uses-hms-as-a-catalog

**Reply by nicor88** [reply]
@alberttwong implementation wise I believe that the feature request that you linked here is similar. The based on the catalog type there could be some specific "catalog" implementation (for example for glue), but the all feature holds the same concept:
the catalog is the "single source of truth" for locations in the object storage, and as so, I believe that Motherduck should support it at some point.

**Comment by alberttwong** [comment]
https://motherduckcommunity.slack.com/archives/C059TEX0189/p1720034758878409

**Comment by mariotaddeucci** [comment]
https://duckdb.org/2025/03/14/preview-amazon-s3-tables

**Comment by bgupta** [comment]
This would be incredibly helpful to me. It would greatly ease migration of Athena code to DuckDB. (I would probably refactor my code to work with both.)

I don’t know if everyone is aware but the Glue Catalog is basically hive compatible.

**Comment by szarnyasg** [comment]
This has experimental support since March 2025: https://duckdb.org/docs/stable/core_extensions/iceberg/amazon_sagemaker_lakehouse.html

**Comment by yokasaka** [comment]
Does it support the GlueDataCatalog? (not using s3tables)

**Comment by zviklausner** [comment]
I tried the experimental feature suggested above - I'm not using iceberg tables in my catalog, just regular parquet tables.
I was able to attach to glue, and in the Motherduck ui it displayed nicely all the schemas AND tables - but whenever I query a table it fails with ```Catalog Error: Table with name ... does not exist!```
